lib/http_server.js

```diff
@@ -446,10 +446,34 @@ ZettaHttpServer.prototype.proxyToPeer = function(env, next) {
     });
 
     res.statusCode = response.statusCode;
-    var buf = [];
+
     if(!opts.pipe) {
-      response.on('data', function(d) {
-        buf += d;  
+      var body = null;
+      var buf = [];
+      var len = 0;
+
+      response.on('readable', function() {
+        var chunk;
+
+        while ((chunk = response.read()) != null) {
+          buf.push(chunk);
+          len += chunk.length;
+        }
+
+        if (!buf.length) {
+          return;
+        }
+
+        if (buf.length && Buffer.isBuffer(buf[0])) {
```

Comment:
Which it looks like it came from argo https://github.com/argo/argo/blob/9e39f7089a367f156325c2e727c6d5f89ba1b710/argo.js#L66-L91.